hoodie-client/src/main/java/com/uber/hoodie/io/compact/HoodieRealtimeTableCompactor.java

```diff
@@ -114,6 +114,9 @@ public HoodieCompactionMetadata compact(JavaSparkContext jsc, HoodieWriteConfig
     for (CompactionWriteStat stat : updateStatusMap) {
       metadata.addWriteStat(stat.getPartitionPath(), stat);
     }
+    // Update file ID and relative paths in commit metadata
+    metadata.updateFileIdAndPaths();
```

Comment:
We should need to be doing this once you change not to serialize the relative paths